src/tests/presets/test_erc721.cairo

```diff
@@ -0,0 +1,1119 @@
+
+    testing::set_contract_address(OPERATOR());
+    dispatcher.safe_transfer_from(owner, receiver, token_id, DATA(true));
+    assert_event_transfer(dispatcher.contract_address, owner, receiver, token_id);
+
+    assert_state_after_transfer(dispatcher, owner, receiver, token_id);
+}
+
+#[test]
+#[available_gas(20000000)]
+fn test_safeTransferFrom_approved_for_all_camel() {
+    let dispatcher = setup_dispatcher();
+    let receiver = setup_camel_receiver();
+    let token_id = TOKEN_1;
+    let owner = OWNER();
+
+    assert_state_before_transfer(dispatcher, owner, receiver, token_id);
+
+    testing::set_contract_address(owner);
+    dispatcher.set_approval_for_all(OPERATOR(), true);
+    utils::drop_event(dispatcher.contract_address);
+
+    testing::set_contract_address(OPERATOR());
+    dispatcher.safeTransferFrom(owner, receiver, token_id, DATA(true));
+    assert_event_transfer(dispatcher.contract_address, owner, receiver, token_id);
+
+    assert_state_after_transfer(dispatcher, owner, receiver, token_id);
+}
+
+#[test]
+#[available_gas(20000000)]
+#[should_panic(expected: ('ERC721: unauthorized caller', 'ENTRYPOINT_FAILED'))]
+fn test_safe_transfer_from_unauthorized() {
+    let dispatcher = setup_dispatcher();
+    testing::set_contract_address(OTHER());
+    dispatcher.safe_transfer_from(OWNER(), RECIPIENT(), TOKEN_1, DATA(true));
+}
+
+#[test]
+#[available_gas(20000000)]
+#[should_panic(expected: ('ERC721: unauthorized caller', 'ENTRYPOINT_FAILED'))]
+fn test_safeTransferFrom_unauthorized() {
+    let dispatcher = setup_dispatcher();
+    testing::set_contract_address(OTHER());
+    dispatcher.safeTransferFrom(OWNER(), RECIPIENT(), TOKEN_1, DATA(true));
+}
+
+//
+// Helpers
+//
+
+fn assert_state_before_transfer(
+    dispatcher: ERC721ABIDispatcher,
+    owner: ContractAddress,
+    recipient: ContractAddress,
+    token_id: u256
+) {
+    assert(dispatcher.owner_of(token_id) == owner, 'Ownership before');
+    assert(dispatcher.balance_of(owner) == TOKENS_LEN, 'Balance of owner before');
+    assert(dispatcher.balance_of(recipient) == 0, 'Balance of recipient before');
+}
+
+fn assert_state_after_transfer(
+    dispatcher: ERC721ABIDispatcher,
+    owner: ContractAddress,
+    recipient: ContractAddress,
+    token_id: u256
+) {
+    assert(dispatcher.owner_of(token_id) == recipient, 'Ownership after');
+    assert(dispatcher.balance_of(owner) == TOKENS_LEN - 1, 'Balance of owner after');
+    assert(dispatcher.balance_of(recipient) == 1, 'Balance of recipient after');
+    assert(dispatcher.get_approved(token_id) == ZERO(), 'Approval not implicitly reset');
+}
+
+fn assert_state_transfer_to_self(
+    dispatcher: ERC721ABIDispatcher, target: ContractAddress, token_id: u256, token_balance: u256
+) {
+    assert(dispatcher.owner_of(token_id) == target, 'Ownership before');
+    assert(dispatcher.balance_of(target) == token_balance, 'Balance of owner before');
+}
+
+fn assert_event_approval_for_all(
```

Comment:
We can import the helpers that are equal from the ERC721 component tests.

Reply:
Unfortunately, none of the helpers are equal. And I tried adding the contract address as a param in the component event tests to see if we could reuse them, but it raised the gas by a significant amount and crashed the tests :(